go/common/version/version.go

```diff
@@ -47,9 +47,9 @@ func (v Version) ToU64() uint64 {
 // FromU64 returns the version from platform-dependent uint64.
 func FromU64(v uint64) Version {
 	return Version{
-		Major: uint16((v >> 32) & 0xff),
-		Minor: uint16((v >> 16) & 0xff),
-		Patch: uint16(v & 0xff),
+		Major: uint16((v >> 32) & 0xffff),
```

Comment:
Please add unit tests to make sure this conversion works for versions that use full uint16 components (e.g. >255).